### PR TITLE
#230 Add --trust parameter to FETCH and PUSH

### PR DESCRIPTION
**Pull request by mgrassotti**
#230 

I think I found a solution but tests are still not working properly.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#368](https://github.com/zold-io/zold/issues/368) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#368](https://github.com/zold-io/zold/issues/368) is assigned to @ledestin[/z](https://www.0crat.com/u/ledestin), here is [why](https://www.0crat.com/footprint/CAZPZR9FS/1901303793); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by ledestin** [comment]
@mgrassotti please reference the issue in the PR description

**Comment by mgrassotti** [comment]
@ledestin it's in the title: #230

**Comment by ledestin** [comment]
@mgrassotti thanks. it's a QA requirement I think. You can't click it in the title too.

File: lib/zold/commands/push.rb [hunk 1]
```diff
@@ -73,6 +76,11 @@ def push(wallet, opts)
         nodes += 1
         total += push_one(wallet, r, opts)
         done += 1
+        if total >= opts['trust']
```
Comment (ledestin) [inline lib/zold/commands/push.rb]: @mgrassotti looks like a candidate for DRYing.

File: test/commands/test_fetch.rb [hunk 2]
```diff
@@ -64,4 +64,45 @@ def test_fetches_wallet
...
+          'body': File.read(wallet.path),
+          'mtime': Time.now.utc.iso8601
+        }.to_json
+      )
+      remotes = home.remotes
+      remotes.add('localhost', 80)
+      remotes.add('localhost', 81)
+      copies = home.copies(wallet)
+
+      # it fetches only 1 remote if first remote trust is >= --trust
```
Comment (ledestin) [inline test/commands/test_fetch.rb]: @mgrassotti I think it'd look better as two tests. You could convert comments into the test names. The same applies to the push test.

**Comment by ledestin** [comment]
@mgrassotti please see my comments

**Comment by mgrassotti** [comment]
@ledestin I tried to address your comments

File: lib/zold/commands/fetch.rb [hunk 3]
```diff
@@ -124,5 +128,11 @@ def age(json)
         "#{(sec / 3600).round}h"
       end
     end
+
+    def enough_trust?(total, trust_value)
```
Comment (ledestin) [inline lib/zold/commands/fetch.rb]: @mgrassotti please add a puzzle to DRY it later or extract somewhere. The same method exists in FETCH and PUSH.

**Comment by ledestin** [comment]
@mgrassotti one comment, otherwise all good. And please take care of the CI fails.

**Comment by 0crat** [comment]
@ledestin[/z](https://www.0crat.com/u/ledestin) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by ledestin** [comment]
@mgrassotti ping

**Comment by 0crat** [comment]
Code review was too long (304 days), architects (@yegor256) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @ledestin[/z](https://www.0crat.com/u/ledestin), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#368](https://github.com/zold-io/zold/issues/368) is now out of scope

**Comment by 0crat** [comment]
Pull request [#368](https://github.com/zold-io/zold/issues/368) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @ledestin[/z](https://www.0crat.com/u/ledestin)

**Pull request closed without merging**